module/userProjects/helper.js

```diff
@@ -1952,7 +1983,18 @@ module.exports = class UserProjectsHelper {
             
             let totalCount = 0;
             let data = [];
-            
+            //this is to check if user has already joined program or not
+            let programJoined = await programUsers.programUsersDocument({userId: userId},["programId"])
```

Comment:
@ankitpws pls clarify why do we need this code in both userAssigned and details(template) functions? front-end is always calling **userProjects/details** api to fetch project details.

Reply:
Is this comment addressed?

Reply:
We need this validation solution details API